generate_resource.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
```

Comment:
Suggest to use the user's environment to determine the location of bash because it's not always located in /bin:

```suggestion
#!/usr/bin/env bash
```